pickhardtpayments/UncertaintyNetwork.py

```diff
@@ -100,15 +105,51 @@ def activate_foaf_uncertainty_reduction(self, src, dest):
             key = "{}x{}".format(vals[0], vals[1])
             if key in ego_network:
                 l = arc.get_actual_liquidity()
-                arc.update_knowledge(l-1)
-                arc.update_knowledge(l+1)
+                arc.update_knowledge(l - 1)
+                arc.update_knowledge(l + 1)
                 # print(key,arc.entropy())
 
             if key in foaf_network:
                 arc.learn_n_bits(2)
                 l = arc.get_actual_liquidity()
-                arc.update_knowledge(l-1)
-                arc.update_knowledge(l+1)
+                arc.update_knowledge(l - 1)
+                arc.update_knowledge(l + 1)
                 # print(key, arc.entropy())
         print("channels with full knowledge: ", len(ego_network))
         print("channels with 2 Bits of less entropy: ", len(foaf_network))
+
+    def settle_attempt(self, attempt: Attempt):
+        """
+        receives a payment attempt and adjusts the balances of the UncertaintyChannels and its reverse channels
+        along the path.
+        """
+        for channel in attempt.path:
+            return_channel = self.get_channel(channel.dest,
+                                              channel.src,
+                                              channel.short_channel_id)
+
+            if channel.min_liquidity < attempt.amount:
+                raise ValueError
+            channel.min_liquidity = max(0, channel.min_liquidity - attempt.amount)
+            if return_channel:
+                return_channel.min_liquidity = min(max(return_channel.min_liquidity, attempt.amount), return_channel.capacity)
```

Comment:
why is attmpt amount not added to return_channel.min_liqudity?

Reply:
This might be a conservative strategy, but the return_channel's minimum liquidity could have been increased in other attempts and already be higher than the amount. Increasing by the amount might overshoot the minimum. For this reason I only check if minimum liquidity is at least the amount.